code/core/src/csf-tools/vitest-plugin/transformer.test.ts

```diff
@@ -0,0 +1,310 @@
+          component: Button
+        };
+        export default _meta;
+      `);
+    });
+
+    it('should add title to const declared default export in story file if not present', async () => {
+      const code = `
+        const meta = {
+          component: Button,
+        };
+  
+        export default meta;
+      `;
+
+      const result = await transform({ code });
+
+      expect(getStoryTitle).toHaveBeenCalled();
+
+      expect(result.code).toMatchInlineSnapshot(`
+        import { test as _test } from "vitest";
+        import { composeStory as _composeStory } from "storybook/internal/preview-api";
+        import { testStory as _testStory, isValidTest as _isValidTest } from "@storybook/experimental-addon-vitest/internal/test-utils";
+        const meta = {
+          component: Button,
+          title: "automatic/calculated/title"
+        };
+        export default meta;
+      `);
+    });
+
+    it('should NOT add title to const declared default export in story file if already present', async () => {
+      const code = `
+        const meta = {
+          title: 'Button',
+          component: Button,
+        };
+  
+        export default meta;
+      `;
+
+      const result = await transform({ code });
+
+      expect(getStoryTitle).not.toHaveBeenCalled();
+
+      expect(result.code).toMatchInlineSnapshot(`
+        import { test as _test } from "vitest";
+        import { composeStory as _composeStory } from "storybook/internal/preview-api";
+        import { testStory as _testStory, isValidTest as _isValidTest } from "@storybook/experimental-addon-vitest/internal/test-utils";
+        const meta = {
+          title: 'Button',
+          component: Button
+        };
+        export default meta;
+      `);
+    });
+  });
+
+  describe('named exports (stories)', () => {
+    it('should add test statement to inline exported stories', async () => {
+      const code = `
+      export default {
+        component: Button,
+      }
+      export const Primary = {
+        args: {
+          label: 'Primary Button',
+        },
+      };
+    `;
+
+      const result = await transform({ code });
+
+      expect(result.code).toMatchInlineSnapshot(`
+        import { test as _test } from "vitest";
+        import { composeStory as _composeStory } from "storybook/internal/preview-api";
+        import { testStory as _testStory, isValidTest as _isValidTest } from "@storybook/experimental-addon-vitest/internal/test-utils";
+        const _meta = {
+          component: Button,
+          title: "automatic/calculated/title"
+        };
+        export default _meta;
+        export const Primary = {
+          args: {
+            label: 'Primary Button'
+          }
+        };
+        const _composedPrimary = _composeStory(Primary, _meta);
+        if (_isValidTest(_composedPrimary, _meta, {"include":[],"exclude":[]})) {
+          _test("Primary", _testStory(_composedPrimary, {"include":[],"exclude":[]}));
+        }
```

Comment:
Not sure if it easy to detect, but we might be able to only set the tags filter, when they are not the default or something?